sdk/cosmos/azure-cosmos-kafka-connect/src/main/java/com/azure/cosmos/kafka/connect/implementation/source/JsonToStruct.java

```diff
@@ -0,0 +1,206 @@
+            nodeSchema.fields().forEach(field -> {
+                JsonNode fieldValue = node.get(field.name());
+                if (fieldValue != null) {
+                    SchemaAndValue schemaAndValue = toSchemaAndValue(field.schema(), fieldValue);
+                    struct.put(field, schemaAndValue.value());
+                } else {
+                    boolean optionalField = field.schema().isOptional();
+                    Object defaultValue = field.schema().defaultValue();
+                    if (optionalField || defaultValue != null) {
+                        struct.put(field, defaultValue);
+                    } else {
+                        logger.error("Missing value for field {}", field.name());
+                    }
+                }
+            });
+        }
+        return new SchemaAndValue(nodeSchema, struct);
+    }
+
+    private static Schema inferSchema(JsonNode jsonNode) {
+        switch (jsonNode.getNodeType()) {
+            case NULL:
+                return Schema.OPTIONAL_STRING_SCHEMA;
+            case BOOLEAN:
+                return Schema.BOOLEAN_SCHEMA;
+            case NUMBER:
+                if (jsonNode.isIntegralNumber()) {
+                    return Schema.INT64_SCHEMA;
+                } else {
+                    return Schema.FLOAT64_SCHEMA;
+                }
+            case ARRAY:
+                List<JsonNode> jsonValues = new ArrayList<>();
+                SchemaBuilder arrayBuilder;
+                jsonNode.forEach(jn -> jsonValues.add(jn));
+
+                Schema firstItemSchema = jsonValues.isEmpty() ? Schema.OPTIONAL_STRING_SCHEMA
+                    : inferSchema(jsonValues.get(0));
+                if (jsonValues.isEmpty() || jsonValues.stream()
+                    .anyMatch(jv -> !Objects.equals(inferSchema(jv), firstItemSchema))) {
+                    // If array is emtpy or it contains elements with different schema types
+                    arrayBuilder = SchemaBuilder.array(Schema.OPTIONAL_STRING_SCHEMA);
+                    arrayBuilder.name(generateName(arrayBuilder));
+                    return arrayBuilder.optional().build();
+                }
+                arrayBuilder = SchemaBuilder.array(inferSchema(jsonValues.get(0)));
+                arrayBuilder.name(generateName(arrayBuilder));
+                return arrayBuilder.optional().build();
+            case OBJECT:
+                SchemaBuilder structBuilder = SchemaBuilder.struct();
+                Iterator<Map.Entry<String, JsonNode>> it = jsonNode.fields();
+                while (it.hasNext()) {
+                    Map.Entry<String, JsonNode> entry = it.next();
+                    structBuilder.field(entry.getKey(), inferSchema(entry.getValue()));
+                }
+                structBuilder.name(generateName(structBuilder));
+                return structBuilder.build();
+            case STRING:
+                return Schema.STRING_SCHEMA;
+            case BINARY:
```

Comment:
do we need to eventually support binary and Pojo. What does Pojo mean?

Reply:
Added a TODO[GA] comment. Not sure how it is used in the past, will revisit before GA